### PR TITLE
Attribute table - Fix the export when a selection or a filter is active

### PR DESCRIPTION
**Pull request by mdouchin**
* when the layer is not spatial
* when the layer has parenthesis inside its name

* Ticket : #2715
* Funded by 3liz

@rldhont need to check if I need to backport it to 3.4

### CODE REVIEW

**Comment by rldhont** [comment]
Regression exists in 3.4

**Review by rldhont — APPROVED** [review]

File: lizmap/www/assets/js/map.js [hunk 1]
```diff
@@ -4724,6 +4724,24 @@ var lizMap = function() {
                   setTimeout(function () { URL.revokeObjectURL(downloadUrl); }, 100); // cleanup
               }
           }
+
+          // REMOVE WHEN THE QGIS SERVER BUG HAS BEEN FIXED
```
Comment (Gustry) [inline lizmap/www/assets/js/map.js:4728]:
```suggestion
          // Note 31/01/2022
          // REMOVE WHEN THE QGIS SERVER BUG HAS BEEN FIXED
          // Related PR for QGIS Master https://github.com/qgis/QGIS/pull/47051
          // It should be fixed for 3.24.1 and 3.22.5
```

File: tests/end2end/cypress/integration/feature_toolbar-ghaction.js [hunk 2]
```diff
@@ -146,15 +146,231 @@ describe('Feature Toolbar', function () {
...
+        cy.wait('@GetExport')
+        .then(({response}) => {
+            cy.fixture('export/export_parent_layer_feature_2.geojson').then((fixtureGeoJSON) => {
+                expect(response.statusCode).to.eq(200)
+                expect(response.body).to.deep.eq(JSON.parse(fixtureGeoJSON))
+            })
+        })
+
+        // Unfilter
+        cy.get('#attribute-layer-main-parent_layer .btn-filter-attributeTable').click({ force: true })
```
Comment (nboisteault) [inline tests/end2end/cypress/integration/feature_toolbar-ghaction.js:247]: Is it needed to unfilter as page get reloaded?

File: tests/end2end/cypress/integration/feature_toolbar-ghaction.js [hunk 3]
```diff
@@ -146,15 +146,231 @@ describe('Feature Toolbar', function () {
...
+        cy.wait('@GetExport')
+        .then(({response}) => {
+            cy.fixture('export/export_data_uids_features_2_and_4.geojson').then((fixtureGeoJSON) => {
+                expect(response.statusCode).to.eq(200)
+                expect(response.body).to.deep.eq(JSON.parse(fixtureGeoJSON))
+            })
+        })
+
+        // Unfilter
+        cy.get('#attribute-layer-main-data_uids .btn-filter-attributeTable').click({ force: true })
```
Comment (nboisteault) [inline tests/end2end/cypress/integration/feature_toolbar-ghaction.js:323]: Is it needed to unfilter as page get reloaded?

**Review by nboisteault — APPROVED** [review]
LGTM. I'll fix `should start child edition linked to a parent feature` test after you merge.

File: tests/end2end/cypress/integration/feature_toolbar-ghaction.js (hunk 2 above)
Comment (mdouchin) [reply tests/end2end/cypress/integration/feature_toolbar-ghaction.js:247]: nope, will remove it

**Review by rldhont — CHANGES_REQUESTED** [review]
Update https://github.com/3liz/lizmap-web-client/blob/release_3_5/lizmap/modules/lizmap/lib/Form/QgisFormValueRelationDynamicDatasource.php#L73

In the backport 3.4 it will be here https://github.com/3liz/lizmap-web-client/blob/release_3_4/lizmap/modules/lizmap/classes/qgisFormValueRelationDynamicDatasource.class.php#L72

**Comment by mdouchin** [comment]
> Update https://github.com/3liz/lizmap-web-client/blob/release_3_5/lizmap/modules/lizmap/lib/Form/QgisFormValueRelationDynamicDatasource.php#L73
> 
> In the backport 3.4 it will be here https://github.com/3liz/lizmap-web-client/blob/release_3_4/lizmap/modules/lizmap/classes/qgisFormValueRelationDynamicDatasource.class.php#L72

You mean I need to change also this part of the code to use my new method  `getWfsTypeName` ?

**Comment by rldhont** [comment]
> > Update https://github.com/3liz/lizmap-web-client/blob/release_3_5/lizmap/modules/lizmap/lib/Form/QgisFormValueRelationDynamicDatasource.php#L73
> > In the backport 3.4 it will be here https://github.com/3liz/lizmap-web-client/blob/release_3_4/lizmap/modules/lizmap/classes/qgisFormValueRelationDynamicDatasource.class.php#L72
> 
> You mean I need to change also this part of the code to use my new method `getWfsTypeName` ?

I mean, it will be great to do it in this PR to be sure to do not loose it.

**Comment by mdouchin** [comment]
@nboisteault @rldhont PR updated . Can be merged I presume

**Comment by rldhont** [comment]
@mdouchin Go for me!

**Pull request merged**